modules/packages/DistributedDeque.chpl

```diff
@@ -705,168 +762,300 @@ class DistributedDeque : Collection {
         node = node.next;
       }
 
-      slot.lock$;
+      followThis.lock$;
     }
-  }
 
-  iter these(param order : Ordering = Ordering.NONE) : eltType where order == Ordering.FIFO {
-    var localThis = getPrivatizedThis;
+    pragma "no doc"
+    proc Destroy() {
+      for slot in slots do delete slot;

```

Comment:
@mppf 

Hey Michael, can you give me some feedback on this. Would this potentially cause an issue since about `maxTaskPar * (numLocales - 1)` will be remote, hence resulting in `on` statements? I comment this out and it works fine for 'fifo', although there is nothing inherently wrong here (as we do not acquire any locks) and this is done sequentially, so at most we only have 2 tasks running...

Reply:
I don't see anything wrong with it either...